csharp/src/Ice/SlicTransceiver.cs

```diff
@@ -376,14 +413,19 @@ internal void FinishedReceivedStreamData(long streamId, int frameOffset, int fra
 
         internal Task PrepareAndSendFrameAsync(
             SlicDefinitions.FrameType type,
-            Func<OutputStream, long>? writer,
+            Action<OutputStream>? writer = null,
+            long? streamId = null,
```

Comment:
Can't we use `ulong` here too as we do for marshaling?

Reply:
We choose to map stream IDs to long in C# even though it's encoded as varulong so we use long in the API.